apps/labs/posts/intel-oneapi-msvc-spin-scipy.md

```diff
@@ -0,0 +1,88 @@
+
+Now coming on to the `arpack` issue. Fixing this issue required supporting building SciPy with Intel oneAPI. On Linux the `arpack` issue was already resolved. However, there were some other issues with `ifx`. For example, `test_equal_bounds` test in `scipy/optimize/tests/test_optimize.py` failed due to floating point issues with `ifx`. Also tolerances for other tests had to be increased (i.e., slight lesser accuracy/precision in order to make things with `ifx`). All of this was done in `BUG/CI: Compile and run tests with `ifx`+`MKL` on Linux` [[gh-21173]](https://github.com/scipy/scipy/pull/21173). I also added a CI job in this PR with `gcc` + `g++` +`ifx` + `MKL` combination. A second steps towards supporting Intel oneAPI with SciPy.
+
+The final checkpoint for Intel oneAPI was `CI: Test icx + icpx + ifx + MKL build of SciPy` [[gh-21254]](https://github.com/scipy/scipy/pull/21254). Here we replaced `gcc` with `icx` and `g++` with `icpx` and TADA, it worked on Linux. Regarding Windows, we are still at `MSVC` + `ifx` + `OpenBLAS` combination because of the `arpack` import error. We tried several ways to fix it but all resulted in a dead end. We tried to build NumPy with `icx`, `icpx` but `icx` is unable to compile NumPy code as is. Some workarounds had to be made. However, still it didn't work for us.
+
+#### Moving away from `dev.py` and migrating to `spin`
+
+Before diving into the details of this work, I would first share some information (for context) related to `meson`. Basically,`meson` builds a project in three steps,
+
+1. First step is that it calls `meson setup` which does configuration for building the project. If you are a CMake user then its just like calling, `cmake .` in a project.
+2. Second step involves calling `meson compile` which executes the compiler commands and creates libraries. Again if you are `cmake` + `make` user then its like calling, `make` or `make -j8`.
+3. Third step is finally calling, `meson install` which installs the libraries, source files (for python) in the installation directory.
+
+Now it would be easy to understand our work below.
+
+We decided to shift to `spin` because of the several issues reported in `dev.py`. For example, [doit interfering with command history of pdb] [[gh-16452]](https://github.com/scipy/scipy/issues/16452), usage of emojis giving issues in a Windows CI environment [[gh-18046]](https://github.com/scipy/scipy/issues/18046), and dev.py unable to resolve homebrew installed Python [[gh-18998]](https://github.com/scipy/scipy/issues/18998). A bunch of those are due to `doit` being a dependency of `dev.py`. One nice feature of `spin` is it has lesser dependencies. This means, lesser bugs because more dependencies means more points of failure. The tradeoff here is to compromise looks and feel of the output a little bit to avoid problems while developing SciPy.
```

Comment:
repeating grammar issue, so commenting: it's "fewer bugs" rather than "lesser bugs" (check for "lesser" in the whole post please)